src/solvers/strings/string_refinement.cpp

```diff
@@ -790,6 +790,13 @@ decision_proceduret::resultt string_refinementt::dec_solve()
   for(const auto &instance : initial_instances)
     add_lemma(instance);
 
+  // All generated string should have positive length
+  for(const auto &string : generator.array_pool.created_strings())
+  {
+    add_lemma(binary_relation_exprt{
+      string.length(), ID_ge, from_integer(0, string.length().type())});
+  }
```

Comment:
Commit message should be changed to `Constrain all generated strings to have non-negative length`. I was initially confused because I thought you were excluding 0.

Reply:
I'm massively surprised we didn't already have that constraint somewhere though...

Reply:
A similar constraint exists in the goto-program for all non-deterministic strings generated there. But we didn't have it for new strings generated by the solver itself.